WordPress/gutenberg
# [Mobile] - Remove usage of ColorEdit in the Button block

## Description
After merging the Global styles implementation https://github.com/WordPress/gutenberg/pull/30544 the [colors hook was enabled](https://github.com/WordPress/gutenberg/pull/30544/files#diff-4a9724d1b978e8fc73857762c4be528d6658524ddf40e5f08c9bed555133b28eR237) for mobile. Because of this change, the gradients option was missing in the color settings and for the Button block, the color settings are duplicated. This is because the block was using `ColorEdit` which is not needed when using the color hook. I'm removing this file since it was only being used for mobile since web stopped using it a while back.

This PR also adds a `gradients` support check when registering the blocks, by only allowing this feature for blocks in `ALLOWED_BLOCKS_GRADIENT_SUPPORT`. Ideally, we would have to think of a way to support gradients for all blocks but for now, this is only available for the Button block.

## How has this been tested?

### Test case 1
- Open the demo app with metro running
- Add a Buttons block
- Open the settings of the Button block
- Expect to see the color settings only once (not duplicated)
- Select the background color setting
- **Expect** to see the Gradients option

### Test case 2 (gradients are not visible for other blocks)
- Open the demo app with metro running
- Add a Group block
- Open the settings of the Group block
- Tap on the background color setting
- **Expect to not see** the gradients option available

## Screenshots 
Before|After
-|-
<kbd><img src="https://user-images.githubusercontent.com/4885740/124112828-024eb300-da6b-11eb-80e1-911b89939a6f.gif" width="200" /></kbd> | <kbd><img src="https://user-images.githubusercontent.com/4885740/124112833-037fe000-da6b-11eb-8197-fe310ec2911f.gif" width="200" /></kbd>

## Types of changes
Bug fix

## Checklist:
